src/azure-cli/azure/cli/command_modules/acr/tests/latest/test_acr_commands.py

```diff
@@ -340,3 +340,169 @@ def test_acr_create_with_audits(self, resource_group):
+        self.cmd('acr identity remove --name {registry_name} --identities {identity_id}', self.check('identity', None))
+
+        # add multiple identities
+        result = self.cmd('acr identity assign --name {registry_name} --identities {system_identity} {identity_id}',
+                          self.check('identity.type', 'SystemAssigned, UserAssigned')).get_output_in_json()
+        self.assertUserIdentitiesExpected([self.kwargs['identity_id'].lower()], result['identity'])
+        # add another user identity to existing
+        result = self.cmd('acr identity assign --name {registry_name} --identities {second_identity_id}',
+                          self.check('identity.type', 'SystemAssigned, UserAssigned')).get_output_in_json()
+        self.assertUserIdentitiesExpected([self.kwargs['identity_id'].lower(), self.kwargs['second_identity_id'].lower()], result['identity'])
+
+        # remove identities and validate result
+        self.cmd('acr identity remove --name {registry_name} --identities {second_identity_id}', self.check('identity.type', 'SystemAssigned, UserAssigned'))
+
+        self.cmd('acr identity remove --name {registry_name} --identities {identity_id}', self.check('identity.type', 'SystemAssigned'))
+        self.cmd('acr identity remove --name {registry_name} --identities {system_identity}', self.check('identity', None))
+
+    @ResourceGroupPreparer(location='centraluseuap')
+    def test_acr_with_private_endpoint(self, resource_group):
+        self.kwargs.update({
+            'registry_name': self.create_random_name('testreg', 20),
+            'vnet_name': self.create_random_name('testvnet', 20),
+            'subnet_name': self.create_random_name('testsubnet', 20),
+            'endpoint_name': self.create_random_name('priv_endpoint', 25),
+            'endpoint_conn_name': self.create_random_name('priv_endpointconn', 25),
+            'second_endpoint_name': self.create_random_name('priv_endpoint', 25),
+            'second_endpoint_conn_name': self.create_random_name('priv_endpointconn', 25),
+            'description_msg': 'somedescription'
+        })
+
+        # create subnet with disabled endpoint network policies
+        self.cmd('network vnet create -g {rg} -n {vnet_name} --subnet-name {subnet_name}')
+        self.cmd('network vnet subnet update -g {rg} --vnet-name {vnet_name} --name {subnet_name} --disable-private-endpoint-network-policies true')
+
+        result = self.cmd('acr create --name {registry_name} --resource-group {rg} --sku premium').get_output_in_json()
+        self.kwargs['registry_id'] = result['id']
+
+        # add an endpoint and approve it
+        result = self.cmd('network private-endpoint create -n {endpoint_name} -g {rg} --subnet {subnet_name} --vnet-name {vnet_name}  '
+                          '--private-connection-resource-id {registry_id} --group-ids registry --connection-name {endpoint_conn_name} --manual-request').get_output_in_json()
```

Comment:
I assume there should be a command called `az acr private-link-resource list` to list the valid `group-id`s, such as the `registry` appearing here.

Reply:
@yugangw-msft @adewaleo I think this command is necessary for users to create a private-endpoint. Is there any specific reason?

Reply:
@MyronFanQiu, agreed, please review https://github.com/Azure/azure-cli/pull/12454